python-checks/src/main/java/org/sonar/python/checks/SkLearnEstimatorDontInitializeEstimatedValuesCheck.java

```diff
@@ -0,0 +1,153 @@
+    functionDef.body().accept(visitor);
+    var offendingVariables = visitor.qualifiedExpressions;
+    var secondaryLocation = classDef.name();
+    offendingVariables
+      .forEach((qualifiedExpression, assignmentStatement) -> {
+        var quickFixOptional = createQuickFix(assignmentStatement);
+        var issue = subscriptionContext.addIssue(qualifiedExpression.name(), MESSAGE).secondary(secondaryLocation, MESSAGE_SECONDARY);
+        quickFixOptional.ifPresent(issue::addQuickFix);
+      });
+
+  }
+
+  private static Optional<PythonQuickFix> createQuickFix(AssignmentStatement assignmentStatement) {
+
+    var builder = PythonQuickFix.newQuickFix(QUICK_FIX_MESSAGE);
+
+    if (assignmentStatement.lhsExpressions().size() != 1 || assignmentStatement.lhsExpressions().stream().anyMatch(expressions -> expressions.expressions().size() != 1)) {
+      return Optional.empty();
+    }
+    builder.addTextEdit(TextEditUtils.removeStatement(assignmentStatement));
+    if (assignmentStatement.assignedValue().is(Tree.Kind.NONE)) {
+      return Optional.of(builder.build());
+    }
+    return Optional.empty();
+  }
+
+  private static class VariableDeclarationEndingWithUnderscoreVisitor extends BaseTreeVisitor {
+
+    private final Map<QualifiedExpression, AssignmentStatement> qualifiedExpressions = new HashMap<>();
+
+    private static boolean isOffendingQualifiedExpression(QualifiedExpression qualifiedExpression) {
+      return !qualifiedExpression.name().name().startsWith("__") && qualifiedExpression.name().name().endsWith("_") && qualifiedExpression.qualifier().is(Tree.Kind.NAME)
+        && "self".equals(((org.sonar.plugins.python.api.tree.Name) qualifiedExpression.qualifier()).name());
+    }
+
+    @Override
+    public void visitAssignmentStatement(AssignmentStatement pyAssignmentStatementTree) {
+      var offendingQualifiedExpressions = pyAssignmentStatementTree.lhsExpressions()
+        .stream()
+        .flatMap(expressionList -> expressionList.expressions().stream())
+        .filter(expression -> expression.is(Tree.Kind.QUALIFIED_EXPR))
+        .map(QualifiedExpression.class::cast);
+
+      var offendingTuples = pyAssignmentStatementTree.lhsExpressions()
+        .stream()
+        .flatMap(expressionList -> expressionList.expressions().stream())
+        .filter(expression -> expression.is(Tree.Kind.TUPLE))
+        .map(Tuple.class::cast)
+        .flatMap(tuple -> tuple.elements().stream())
+        .filter(expression -> expression.is(Tree.Kind.QUALIFIED_EXPR))
+        .map(QualifiedExpression.class::cast);
+
+      var qualifiedExpressionsStream = Stream.concat(
+        offendingQualifiedExpressions, offendingTuples)
+        .filter(VariableDeclarationEndingWithUnderscoreVisitor::isOffendingQualifiedExpression);
+
+      qualifiedExpressionsStream.forEach(qualifiedExpression -> qualifiedExpressions.put(qualifiedExpression, pyAssignmentStatementTree));
+      super.visitAssignmentStatement(pyAssignmentStatementTree);
```

Comment:
I did not see that in the previous iteration, but we could drop the assignment to qualifiedExpressionsStream
```suggestion
      Stream.concat(
        offendingQualifiedExpressions, offendingTuples)
        .filter(VariableDeclarationEndingWithUnderscoreVisitor::isOffendingQualifiedExpression)
        .forEach(qualifiedExpression -> qualifiedExpressions.put(qualifiedExpression, pyAssignmentStatementTree));
      super.visitAssignmentStatement(pyAssignmentStatementTree);
```